project/src/main/webapp/script.js

```diff
@@ -181,8 +184,35 @@ function getSize(frequency,max,min) {
   return size;
 }
 
+/** Returns a map of the trends and their respective sentiment scores. */
+function getTrendBubbleScore(trends) {
+  let score = new Map();
+  trends.forEach(trend => score.set(trend.name,getAverageSentiment(trend.articles)));
+  return score;
+}
+
+/** Returns the average of the sentiment scores of all the articles. */
+function getAverageSentiment(articles) {
+  let avg = articles.reduce((sum, value) => sum + value, 0) / length;
+  return avg.toFixed(1);
+}
+
+/** Assigns and returns a color based on the sentiment value. */
+function getColorGradient() {
+  let sentimentValueCount = 20;  
+  let chromaColor = chroma.scale(['#FED8F7','#C4DDFE']).colors(sentimentValueCount);
+  let colors = new Map();
+  let j = 1;
+  for (let i = 0; i <= chromaColor.length; i++) {
+    colors.set(j.toFixed(1),chromaColor[i]);
+    j -= .1;
+  }
+  trend.color = colors;
+} 
+
 /** Shows preloader on page load and fetches articles. */
 function loadPage() {  
   retrieveArticles(5, trend);
   setTimeout(showPage, 4000);
+  getColorGradient();
```

Comment:
can you do ` trend.color = getColorGradient()`?  It's much preferable for your functions to not have side effects.